File: examples/typescript/mcp_collaboration.ts
/**
 * Awareness Market — Multi-AI Collaboration via MCP
 *
 * Create a workspace, spin up a collaboration session, and interact
 * with the MCP server using the JSON-RPC protocol.
 *
 * Usage:
 *   npm install
 *   npx tsx mcp_collaboration.ts
 */

const BASE_URL = "http://localhost:3001";

async function jsonRpc(
  endpoint: string,
  method: string,
  params: Record<string, unknown>,
  headers: Record<string, string>,
  id = 1
) {
  const resp = await fetch(endpoint, {
    method: "POST",
    headers,
    body: JSON.stringify({ jsonrpc: "2.0", method, params, id }),
  });
  return resp.json();
}

async function main() {
  console.log("Awareness Market — Multi-AI Collaboration Demo\n");

  // 1. Authenticate
  console.log("1. Authenticating...");
  const authResp = await fetch(`${BASE_URL}/api/auth/login`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      email: "demo@example.com",
      password: "demo-password",
    }),
  });

  if (!authResp.ok) {
    console.log("   Auth failed. Create a demo account first: pnpm create-demo-account");
    return;
  }

  const { token } = await authResp.json();
  const headers = {
    Authorization: `Bearer ${token}`,
    "Content-Type": "application/json",
  };

  // 2. Create workspace
  console.log("2. Creating workspace...");
  const wsResp = await fetch(`${BASE_URL}/api/trpc/workspace.create`, {
    method: "POST",
    headers,
    body: JSON.stringify({
      name: "TS Collaboration Demo",
      description: "TypeScript multi-agent demo",
    }),
  });

  if (!wsResp.ok) {
    console.log(`   Workspace creation failed (${wsResp.status})`);
    return;
  }

  const wsData = await wsResp.json();
  const workspaceId = wsData.result?.data?.id;
  console.log(`   Workspace: ${workspaceId}`);

  // 3. Create session with two agents
  console.log("3. Creating collaboration session...");
  const sessionResp = await fetch(
    `${BASE_URL}/api/trpc/agentCollaboration.createSession`,
    {
      method: "POST",
      headers,
      body: JSON.stringify({
        workspaceId,
        agents: [
          { role: "architect", name: "Claude", weight: 1.0 },
          { role: "frontend", name: "v0", weight: 0.7 },
        ],
        task: "Design a real-time dashboard for AI memory visualization",
      }),
    }
  );

  if (!sessionResp.ok) {
    console.log(`   Session creation failed (${sessionResp.status})`);
    return;
  }

  const sessionData = await sessionResp.json();
  const { sessionId, mcpToken } = sessionData.result?.data ?? {};
  console.log(`   Session: ${sessionId}`);
  console.log(`   MCP Token: ${mcpToken?.slice(0, 20)}...`);

  // 4. Interact via MCP JSON-RPC
  const mcpHeaders = {
    Authorization: `Bearer ${mcpToken ?? token}`,
    "Content-Type": "application/json",
    "X-Agent-Role": "architect",
  };

  console.log("\n4. Sharing reasoning via MCP...");
  const shareResult = await jsonRpc(
    `${BASE_URL}/mcp`,
    "tools/call",
    {
      name: "share_reasoning",
      arguments: {
        reasoning:
          "For real-time visualization, we need WebSocket for live updates and Three.js for 3D rendering of memory graphs.",
        decision: "Use Socket.IO + Three.js with React wrapper",
        questions: [
          "Should we support 2D fallback for low-end devices?",
          "What refresh rate for live memory updates?",
        ],
      },
    },
    mcpHeaders
  );
  console.log(`   Result: ${JSON.stringify(shareResult).slice(0, 100)}...`);

  console.log("\n5. Proposing a shared decision...");
  const decisionResult = await jsonRpc(
    `${BASE_URL}/mcp`,
    "tools/call",
    {
      name: "propose_shared_decision",
      arguments: {
        decision: "Use React Three Fiber for 3D memory visualization",
        reasoning:
          "React Three Fiber integrates natively with React component lifecycle, making it easier to manage state updates from WebSocket events.",
        alternatives: [
          "Raw Three.js — more control but harder React integration",
          "D3.js — 2D only, but simpler",
        ],
      },
    },
    mcpHeaders
  );
  console.log(`   Result: ${JSON.stringify(decisionResult).slice(0, 100)}...`);

  console.log("\n6. Fetching collaboration history...");
  const historyResult = await jsonRpc(
    `${BASE_URL}/mcp`,
    "tools/call",
    {
      name: "get_collaboration_history",
      arguments: { filter: "decisions", limit: 5 },
    },
    mcpHeaders
  );
  console.log(`   History: ${JSON.stringify(historyResult).slice(0, 200)}...`);

  console.log("\nDone! Connect Claude Desktop using this MCP config:");
  console.log(
    JSON.stringify(
      {
        mcpServers: {
          "awareness-market": {
            command: "node",
            args: ["./mcp-server/dist/index-enhanced.js"],
            env: { VITE_APP_URL: BASE_URL },
          },
        },
      },
      null,
      2
    )
  );
}

main().catch(console.error);

File: examples/python/mcp_collaboration.py
"""
Awareness Market — Multi-AI Collaboration via MCP

Create a collaboration workspace, add AI agents, and run a collaborative task.

Usage:
    pip install requests
    python mcp_collaboration.py
"""

import requests
import json

BASE_URL = "http://localhost:3001"


def main():
    print("Awareness Market — Multi-AI Collaboration Demo\n")

    # Step 1: Authenticate (get a session token)
    print("1. Authenticating...")
    auth_resp = requests.post(f"{BASE_URL}/api/auth/login", json={
        "email": "demo@example.com",
        "password": "demo-password",
    })
    if not auth_resp.ok:
        print("   Auth failed. Make sure you have a demo account:")
        print("   pnpm create-demo-account")
        return

    token = auth_resp.json().get("token", "")
    headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json"}

    # Step 2: Create a workspace
    print("2. Creating collaboration workspace...")
    ws_resp = requests.post(f"{BASE_URL}/api/trpc/workspace.create", json={
        "name": "Demo Collaboration",
        "description": "Multi-agent demo workspace",
    }, headers=headers)
    if ws_resp.ok:
        workspace = ws_resp.json().get("result", {}).get("data", {})
        ws_id = workspace.get("id", "unknown")
        print(f"   Workspace created: {ws_id}")
    else:
        print(f"   Workspace creation returned {ws_resp.status_code}")
        return

    # Step 3: Create a collaboration session with two agents
    print("3. Creating collaboration session...")
    session_resp = requests.post(f"{BASE_URL}/api/trpc/agentCollaboration.createSession", json={
        "workspaceId": ws_id,
        "agents": [
            {"role": "architect", "name": "Claude", "weight": 1.0},
            {"role": "frontend", "name": "v0", "weight": 0.7},
        ],
        "task": "Design a REST API for a knowledge sharing platform",
    }, headers=headers)
    if session_resp.ok:
        session = session_resp.json().get("result", {}).get("data", {})
        session_id = session.get("sessionId", "unknown")
        mcp_token = session.get("mcpToken", "")
        print(f"   Session: {session_id}")
        print(f"   MCP Token: {mcp_token[:20]}..." if mcp_token else "   (no MCP token)")
    else:
        print(f"   Session creation returned {session_resp.status_code}")
        return

    # Step 4: Share reasoning as an agent
    print("4. Sharing reasoning...")
    mcp_headers = {
        "Authorization": f"Bearer {mcp_token}" if mcp_token else f"Bearer {token}",
        "Content-Type": "application/json",
        "X-Agent-Role": "architect",
    }
    reason_resp = requests.post(f"{BASE_URL}/mcp", json={
        "jsonrpc": "2.0",
        "method": "tools/call",
        "params": {
            "name": "share_reasoning",
            "arguments": {
                "reasoning": "We should use RESTful conventions with versioned endpoints.",
                "decision": "Use /api/v1/ prefix for all endpoints",
                "questions": ["Should we support GraphQL as well?"],
            },
        },
        "id": 1,
    }, headers=mcp_headers)
    if reason_resp.ok:
        print("   Reasoning shared successfully")
    else:
        print(f"   Share reasoning returned {reason_resp.status_code}")

    # Step 5: Get collaboration history
    print("5. Fetching collaboration history...")
    history_resp = requests.post(f"{BASE_URL}/mcp", json={
        "jsonrpc": "2.0",
        "method": "tools/call",
        "params": {
            "name": "get_collaboration_history",
            "arguments": {"filter": "all", "limit": 10},
        },
        "id": 2,
    }, headers=mcp_headers)
    if history_resp.ok:
        result = history_resp.json()
        print(f"   History entries: {json.dumps(result, indent=2)[:200]}...")
    else:
        print(f"   History fetch returned {history_resp.status_code}")

    print("\nDone! Connect Claude Desktop to this session using the MCP token above.")
    print("See mcp-server/README.md for MCP client configuration.")


if __name__ == "__main__":
    main()
